policy-controller/k8s/index/src/pod.rs

```diff
@@ -68,6 +68,52 @@ pub(crate) fn tcp_port_names(spec: Option<k8s::PodSpec>) -> HashMap<String, Port
     port_names
 }
 
+/// Gets the container probe ports for a Pod.
+///
+/// The result is a mapping for each probe port exposed by a container in the
+/// Pod and the paths for which probes are expected.
+pub(crate) fn get_container_probes(
+    spec: &Option<k8s::PodSpec>,
```

Comment:
it would be preferable to have this take a `&k8s::PodSpec`, since this function does nothing in the optional case.

Reply:
This is similar to `fn port_names` already taking an `Option<k8s::PodSpec>`. Would you want me to change that function too?

Reply:
Changed in [c0216a6](https://github.com/linkerd/linkerd2/pull/8961/commits/c0216a6cb8f74732492ca507501f5bf5d0ad3774)